src/libflate/lib.rs

```diff
@@ -35,13 +34,27 @@
 extern crate libc;
 
 use libc::{c_void, size_t, c_int};
+use std::fmt;
 use std::ops::Deref;
 use std::ptr::Unique;
 use std::slice;
 
+#[derive(Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Error(());
+
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        "decompression error".fmt(f)
+    }
+}
+
+impl Drop for Error {
```

Comment:
You can still do things like `let Error(_) = e`, removing the possibility of changing this to a struct or an enum later on (or just about anything else than a tuple struct).